catalog-rest-service/src/main/resources/ui/src/components/common/non-admin-action/NonAdminAction.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements. See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License. You may obtain a copy of the License at
+
+  * http://www.apache.org/licenses/LICENSE-2.0
+
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import { findByTestId, render } from '@testing-library/react';
+import React from 'react';
+import NonAdminAction from './NonAdminAction';
+
+const mockAuth = {
+  isAdminUser: true,
+  isAuthDisabled: true,
+};
+
+jest.mock('../../../hooks/authHooks', () => ({
+  useAuth: jest.fn(() => mockAuth),
+}));
+
+jest.mock('../popover/PopOver', () => {
+  return jest
+    .fn()
+    .mockImplementation(({ children }: { children: React.ReactNode }) => (
+      <p data-testid="popover">{children}</p>
+    ));
+});
+
+describe('Test AddUsersModal component', () => {
+  it('Component should render', async () => {
+    const { container } = render(
+      <NonAdminAction>
+        <button data-testid="test-button">test</button>
+      </NonAdminAction>
+    );
+
+    const testButton = await findByTestId(container, 'test-button');
+
+    expect(testButton).toBeInTheDocument();
+  });
+
+  it('Should restrict user if criteria does not match', async () => {
+    mockAuth.isAdminUser = false;
+    mockAuth.isAuthDisabled = false;
+
+    const { container } = render(
+      <NonAdminAction title="test popup" trigger="click">
```

Comment:
I see the trigger is set to click but no click is fired using fireEvent, how is this test being checked?

Reply:
trigger value will be pass to popover component, and the tests of that component are cover under its test.